backend/src/controllers/clientCourses/manageAssessment.ts

```diff
@@ -0,0 +1,50 @@
+import { NextFunction, Request, Response } from 'express';
+
+import { AssessmentAction } from 'enums/common';
+import CourseStatus from 'enums/coursesEnums';
+import BadRequestError from 'classes/errors/clientErrors/BadRequestError';
+import { getStatusProvider, updateClientCourseField } from 'db/providers/clientCourseProvider';
+import { COURSE_FILEDS } from 'config/constants';
+import { TAchievments } from 'interfaces/Ientities/Itest';
+
+const manageAssessment = async (
+  req: Request<{ id: string }, unknown, { action: AssessmentAction }>,
+  res: Response<never, { id: string; achievments: TAchievments; results: string }>,
+  next: NextFunction,
+) => {
+  const { id: courseId } = req.params;
+  const { action: assessmentAction } = req.body;
+
+  try {
+    const { status } = await getStatusProvider(courseId);
+
+    if (status !== CourseStatus.assessment) {
+      throw new BadRequestError(`Can not manage assessment while course in status: ${status}.`);
+    }
+
+    let statusToSet: CourseStatus;
+
+    switch (assessmentAction) {
+      case AssessmentAction.approve:
+        statusToSet = CourseStatus.completed;
+        break;
+      case AssessmentAction.decline:
+        statusToSet = CourseStatus.failed;
+        break;
+      default:
+        throw new BadRequestError('Bad assessment action.');
+    }
+
+    await updateClientCourseField(courseId, COURSE_FILEDS.status, statusToSet);
+
+    res.locals.results = `Assessment has been successfully ${
+      statusToSet === CourseStatus.completed ? 'approved' : 'declined'
```

Comment:
Replase statuses with the constant please

Reply:
It's not status actually. Yep, we have `approved` course status, but not `declined` one. This is actually a message that tells user whether assessment was approved or declined. I can only introduce something like `const ASSESSMENT_RESULT` where we will have both `approved` and `declined`.

Reply:
fixed